climakitae/util/utils.py

```diff
@@ -772,90 +772,102 @@ def summary_table(data: xr.Dataset) -> pd.DataFrame:
     return df
 
 
-def convert_to_local_time(data: xr.DataArray, selections) -> xr.DataArray:
+def convert_to_local_time(data: xr.DataArray | xr.Dataset) -> xr.DataArray:
     """
     Convert time dimension from UTC to local time for the grid or station.
 
     Args:
-        data (xarray.DataArray): Input data.
-        selections: DataParameters object containing selection details.
+        data (xarray.DataArray | xr.Dataset): Input data.
 
     Returns:
         xarray.DataArray: Data with converted time coordinate.
     """
 
+    # Only converting hourly data
+    if "frequency" not in data.attrs:
+        # Make a guess at frequency
+        timestep = pd.Timedelta(
+            data.time[1].item() - data.time[0].item()
+        ).total_seconds()
+        match timestep:
+            case 3600:
+                frequency = "hourly"
+            case 86400:
+                frequency = "daily"
+            case _ if timestep > 86400:
+                frequency = "monthly"
+    else:
+        frequency = str(data.attrs["frequency"])
+
     # If timescale is not hourly, no need to convert
-    if selections.timescale in ["monthly", "daily"]:
+    if frequency != "hourly":
         print(
             "You've selected a timescale that doesn't require any timezone shifting, due to its timescale not being granular enough (hourly). Please pass in more granular level data if you want to adjust its local timezone."
         )
         return data
 
-    # 1. Get the time slice from selections
-    start, end = selections.time_slice
+    # Find out if Stations or Gridded type
+    if "data_type" not in data.attrs:
```

Comment:
```suggestion
    if not (data_type := data.attrs.get("data_type", None)):
```
dictionaries support `.get(key, default)` method that makes these kinds of assignments very nice to read and use. And the walrus operator is handy for checking and setting in the same line, so you can remove the try, except block and the `data_type = ...` line below.